src/Game/Jobs/DRG.ts

```diff
@@ -0,0 +1,1069 @@
+				}
+			}
+			this.resources.addResourceEvent({
+				rscType: "STARCROSS_READY",
+				name: "drop starcross outside life of the dragon",
+				fnOnRsc: dropStarcrossOutsideLife,
+				delay: 1,
+			});
+		};
+		dropStarcrossOutsideLife(this.resources.get("STARCROSS_READY"));
+	}
+
+	override jobSpecificAddDamageBuffCovers(node: ActionNode, skill: Skill<PlayerState>): void {
+		if (this.hasResourceAvailable("POWER_SURGE")) {
+			node.addBuff(BuffType.PowerSurge);
+		}
+		if (this.hasResourceAvailable("LANCE_CHARGE")) {
+			node.addBuff(BuffType.LanceCharge);
+		}
+		if (this.hasResourceAvailable("LIFE_OF_THE_DRAGON")) {
+			node.addBuff(BuffType.LifeOfTheDragon);
+		}
+		if (this.hasResourceAvailable("BATTLE_LITANY")) {
+			node.addBuff(BuffType.BattleLitany);
+		}
+		if (this.hasResourceAvailable("ENHANCED_PIERCING_TALON")) {
+			node.addBuff(BuffType.EnhancedPiercingTalon);
+		}
+		if (this.hasResourceAvailable("LIFE_SURGE")) {
+			node.addBuff(BuffType.LifeSurge);
+		}
+	}
+
+	// handle all DRG combo abilities and combo states
+	fixDRGComboState(skillName: ActionKey) {
+		// consume draconian fire
+		this.tryConsumeResource("DRACONIAN_FIRE", true);
+
+		let resType: DRGResourceKey = "DRG_AOE_COMBO_TRACKER";
+		// HANDLE AOE DIFFERENTLY
+		if (AOE_COMBO_SKILLS.includes(skillName)) {
+			// reset chaos and heavens thrust combo trackers
+			this.tryConsumeResource("DRG_CHAOS_COMBO_TRACKER", true);
+			this.tryConsumeResource("DRG_HEAVENS_COMBO_TRACKER", true);
+			if (skillName === "DOOM_SPIKE" || skillName === "DRACONIAN_FURY") {
+				this.tryConsumeResource(resType, true);
+				this.resources.get(resType).gain(1);
+				this.enqueueResourceDrop(resType);
+			} else if (skillName === "SONIC_THRUST") {
+				if (this.resources.get(resType).availableAmount() === 1) {
+					this.resources.get(resType).gain(1);
+					this.enqueueResourceDrop(resType);
+				} else {
+					this.tryConsumeResource(resType, true);
+				}
+			} else if (skillName === "COERTHAN_TORMENT") {
+				this.tryConsumeResource(resType, true);
+			}
+			return;
+		}
+
+		this.tryConsumeResource("DRG_AOE_COMBO_TRACKER", true);
+
+		// true/raiden thrust
+		if (skillName === "TRUE_THRUST" || skillName === "RAIDEN_THRUST") {
+			// reset combos
+			this.tryConsumeResource("DRG_CHAOS_COMBO_TRACKER", true);
+			this.tryConsumeResource("DRG_HEAVENS_COMBO_TRACKER", true);
+
+			// start the two branching combos
+			this.resources.get("DRG_CHAOS_COMBO_TRACKER").gain(1);
+			this.enqueueResourceDrop("DRG_CHAOS_COMBO_TRACKER");
+			this.resources.get("DRG_HEAVENS_COMBO_TRACKER").gain(1);
+			this.enqueueResourceDrop("DRG_HEAVENS_COMBO_TRACKER");
+
+			return;
+		}
+
+		let combo_value = -1;
+		// chaos combo
+		if (CHAOS_COMBO_MAP.has(skillName)) {
+			this.tryConsumeResource("DRG_HEAVENS_COMBO_TRACKER", true);
+			resType = "DRG_CHAOS_COMBO_TRACKER";
+			combo_value = CHAOS_COMBO_MAP.get(skillName)
+				? Number(CHAOS_COMBO_MAP.get(skillName))
+				: -1;
```

Comment:
```suggestion
			combo_value = CHAOS_COMBO_MAP.get(skillName) ?? -1;
```

Using the `Number` constructor is a bit of an anti-pattern, since it tends to hide possible logic errors.